apps/system/js/app_usage_metrics.js

```diff
@@ -671,20 +728,35 @@
     this.relativeStartTime = performance.now();
   }
 
-  UsageData.prototype.getAppUsage = function(app) {
+  /*
+   * Get app usage for the current date
+   */
+  UsageData.prototype.getAppUsage = function(app, dayKey) {
     var usage = this.data.apps[app];
+    dayKey = dayKey || this.getDayKey();
+
+    // We lazily initialize both the per-app and per-day usage maps
     if (!usage) {
-      // If no usage exists for this app, create a new empty object for it.
-      usage = {
+      this.data.apps[app] = usage = {};
+    }
+
+    var dayUsage = usage[dayKey];
+    if (!dayUsage) {
+      dayUsage = usage[dayKey] = {
         usageTime: 0,
         invocations: 0,
         installs: 0,
         uninstalls: 0,
         activities: {}
       };
-      this.data.apps[app] = usage;
     }
-    return usage;
+    return dayUsage;
+  };
```

Comment:
My intuition about this is that this data would be more usefully organized first by date and then by app. But you have the app url as the top-level property and then have dates below that.  I don't think it matters much, and if the server side is already being coded to expect this organization, then I'd leave it alone. But if not, at least consider changing data.apps[app][day] to data.days[day][app].